app/process_proposal.go

```diff
@@ -90,6 +90,9 @@ func (app *App) ProcessProposal(req abci.RequestProcessProposal) (resp abci.Resp
 				return reject()
 			}
 
+			// Set the tx size on the context before calling the AnteHandler
+			sdkCtx = sdkCtx.WithTxBytes(tx)
```

Comment:
[question] does this need to be gated behind a version gate? @evan-forbes mentioned that it is consensus breaking

```suggestion
		// Set the tx bytes in the context for app version v3 and greater
		if sdkCtx.BlockHeader().Version.App >= 3 {
			sdkCtx = sdkCtx.WithTxBytes(tx)
		}
```

Reply:
me and @evan-forbes briefly discussed this yesterday but I don't think we reached a firm conclusion. `ConsumeTxSizeGasDecorator` also uses TxBytes set on the context so I guess it is consensus breaking? I could also reset it back to empty after `MaxTxSizeDecorator` is executed.

Reply:
It seems like a bug that it wasn't set before `ConsumeTxSizeGasDecorator`. 

> do we want it to be set during the ante handler execution in v3?

IMO yes

Reply:
I added this gate in https://github.com/celestiaorg/celestia-app/pull/3960 (which we can close after this)

I think there's a chance it could be consensus breaking because of the ante handle that subtracts gas for the size of the tx. we don't currently set it in process proposal afaik, which would mean that's its possible to reject or accept a block if an account does or doesn't have enough gas. very unlikely, but I think it could happen.

Reply:
so we need a gate

Reply:
makes sense! thanks!

Reply:
https://github.com/celestiaorg/celestia-app/pull/3909/commits/54b2b4d1ae3874579b1617aee8da258be4bd7f4c

Reply:
https://github.com/celestiaorg/celestia-app/pull/3909/commits/a873099ce3d6f205a1bcf5c6768c3a9926852a7f